### ISSUE TITLE
Is it possible to include double quotes in the `cmd` argument?

### ISSUE THREAD

**Issue by reidprichard** [akinsho/toggleterm.nvim#508]
I run Windows, and due to organizational constraints I'm forced to have spaces in my directories. I frequently want to execute the open Python file with `TermExec cmd="python %"`. As you can imagine, this causes issues if `%` expands to a path with a space, as you get e.g.:
> python C:/this/is the/file/path.py
Can't open file 'C:/this/is': no such file or directory.

I've tried every which way to escape quotes, mixing single and double quotes, escaping spaces with backslashes/backticks, etc., but I'm getting nowhere. Using the `dir` argument to `TermExec` works if the terminal is not open yet, but it will not change the working directory if there's an open terminal already.

Thoughts?

**Comment by mooreye** [comment]
I'm looking for a solution to a broader problem, namely I want to execute the currently opened file in terminal:
`:TermExec cmd='./%'`
but this fails if filename contains any shell special characters, not just spaces but also things like `' " $` etc. Is there a way to pass a shell-escaped filename to `cmd=`?

**Comment by reidprichard** [comment]
Can't you just surround the filename in quotes?

**Comment by mooreye** [comment]
> Can't you just surround the filename in quotes?

No, that won't work for cases when `"` is part of the filename, and will also try to expand variables if filename contains something like `$foo`

**Comment by reidprichard** [comment]
Oh, duh. Here's my code to run the current file as a Python script. I don't specially handle anything but spaces, but hopefully this is a useful starting point. I wasn't aware of the `shellescape()` function, but I think you could just switch it out for my `gsub` call.
```lua
local function run_python()
  vim.cmd.write()
  print("Running...")
  local script_path = vim.fn.expand("%:p"):gsub(" ", "\\ ")
  local command = "python \"" .. script_path .. "\""
  toggleterm.exec(command)
end
```

**Comment by jiaxinaoliao** [comment]
> I run Windows, and due to organizational constraints I'm forced to have spaces in my directories. I frequently want to execute the open Python file with `TermExec cmd="python %"`. As you can imagine, this causes issues if `%` expands to a path with a space, as you get e.g.:
> 
> > python C:/this/is the/file/path.py
> > Can't open file 'C:/this/is': no such file or directory.
> 
> I've tried every which way to escape quotes, mixing single and double quotes, escaping spaces with backslashes/backticks, etc., but I'm getting nowhere. Using the `dir` argument to `TermExec` works if the terminal is not open yet, but it will not change the working directory if there's an open terminal already.
> 
> Thoughts?

![image](https://github.com/user-attachments/assets/77023b43-6d4a-4bc6-b5e4-92f94420be00)
windows下pwsh可以用--代替“”

**Comment by jiaxinaoliao** [comment]
> > I run Windows, and due to organizational constraints I'm forced to have spaces in my directories. I frequently want to execute the open Python file with `TermExec cmd="python %"`. As you can imagine, this causes issues if `%` expands to a path with a space, as you get e.g.:
> > > python C:/this/is the/file/path.py
> > > Can't open file 'C:/this/is': no such file or directory.
> > 
> > 
> > I've tried every which way to escape quotes, mixing single and double quotes, escaping spaces with backslashes/backticks, etc., but I'm getting nowhere. Using the `dir` argument to `TermExec` works if the terminal is not open yet, but it will not change the working directory if there's an open terminal already.
> > Thoughts?
> 
> ![image](https://private-user-images.githubusercontent.com/98008397/373667740-77023b43-6d4a-4bc6-b5e4-92f94420be00.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgwNTI1MjgsIm5iZiI6MTcyODA1MjIyOCwicGF0aCI6Ii85ODAwODM5Ny8zNzM2Njc3NDAtNzcwMjNiNDMtNmQ0YS00YmM2LWI1ZTQtOTJmOTQ0MjBiZTAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA0VDE0MzAyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIyMmZmNTQ1Y2MxNjRmNDA1ZjAyZTc1ZThmYjAzNjI1NWRlNWFmZGRiMzY0YmM4OWY3ZDc5MDYzOGEwMDRhOTcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.3uIxqNjGUMW4z-AkHEy-TB4EzdSHBdnvr3bCirzk6Y8) windows下pwsh可以用--代替“”

应该用\替代”
![image](https://github.com/user-attachments/assets/4284a930-4612-470b-bf5d-0d94c5b7219c)

**Comment by jiaxinaoliao** [comment]
> > I run Windows, and due to organizational constraints I'm forced to have spaces in my directories. I frequently want to execute the open Python file with `TermExec cmd="python %"`. As you can imagine, this causes issues if `%` expands to a path with a space, as you get e.g.:
> > > python C:/this/is the/file/path.py
> > > Can't open file 'C:/this/is': no such file or directory.
> > 
> > 
> > I've tried every which way to escape quotes, mixing single and double quotes, escaping spaces with backslashes/backticks, etc., but I'm getting nowhere. Using the `dir` argument to `TermExec` works if the terminal is not open yet, but it will not change the working directory if there's an open terminal already.
> > Thoughts?
> 
> ![image](https://private-user-images.githubusercontent.com/98008397/373667740-77023b43-6d4a-4bc6-b5e4-92f94420be00.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjgwNTMwMDIsIm5iZiI6MTcyODA1MjcwMiwicGF0aCI6Ii85ODAwODM5Ny8zNzM2Njc3NDAtNzcwMjNiNDMtNmQ0YS00YmM2LWI1ZTQtOTJmOTQ0MjBiZTAwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEwMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMDA0VDE0MzgyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg3N2U4MWEzNzQ3NzQ4MDA0MWE4NDY4YjhjNmExNzg5MmZkMDhhNWM2ZDgxMTI4MGY0MGUwZWRmMDRlYmY2M2EmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RS92NWgZLjHQ7AWwAnsLRT7EEWQag-PhoJqXTehtAY4) windows下pwsh可以用--代替“”

应该用.\相对路径代替”
![image](https://github.com/user-attachments/assets/728eced7-8641-4672-b953-7b0d630f14c8)
![image](https://github.com/user-attachments/assets/becb0f30-b258-4f49-99f9-9eefd412da1f)
